app/controllers/results_controller.rb

```diff
@@ -139,7 +139,11 @@ def show
           data[:notes_count] = submission.grouping.notes.count
           data[:num_marked] = assignment.get_num_marked(current_user.admin? ? nil : current_user.id)
           data[:num_assigned] = assignment.get_num_assigned(current_user.admin? ? nil : current_user.id)
-          data[:group_name] = submission.grouping.get_group_name
+          if current_user.ta? && assignment.anonymize_groups
```

Comment:
Elsewhere you use `!current_user.admin?`, so I think you should be consistent here.

Reply:
Or better, use `ActionPolicy` for this? Something like `allowed_to? :view_group_membership, assignment`.

Reply:
Looking at this now I think we should actually change the rest to `current_user.ta?`. It's always clearer to make the positive check. Thoughts?

And yes, eventually I want to do a lot of action policy changes to these files but I figure that's a separate PR.